src/sentry/static/sentry/app/components/tag.tsx

```diff
@@ -1,76 +1,151 @@
+        {defined(onDismiss) && (
+          <DismissButton
+            onClick={handleDismiss}
+            size="zero"
+            priority="link"
+            label={t('Dismiss')}
+          >
+            <IconClose {...iconsProps} />
+          </DismissButton>
+        )}
+      </Background>
+    </Tooltip>
+  );
+
+  function handleDismiss(event: React.MouseEvent) {
+    event.preventDefault();
+    onDismiss?.();
   }
 
-  return null;
-};
+  function tagIcon() {
+    if (React.isValidElement(icon)) {
+      return <IconWrapper>{React.cloneElement(icon, {...iconsProps})}</IconWrapper>;
+    }
 
-const getMarginLeft = (p: StyleFuncProps) =>
-  p.inline ? `margin-left: ${p.size === 'small' ? '0.25em' : '0.5em'};` : '';
-
-const getBorder = (p: StyleFuncProps) =>
-  p.border ? `border: 1px solid ${getPriority(p)?.border ?? p.theme.borderDark};` : '';
-
-const Tag = styled(
-  ({
-    children,
-    icon,
-    inline: _inline,
-    priority: _priority,
-    size: _size,
-    border: _border,
-    ...props
-  }: Props) => (
-    <div {...props}>
-      {icon && (
+    if ((defined(href) || defined(to)) && icon === undefined) {
+      return (
         <IconWrapper>
-          {React.isValidElement(icon) && React.cloneElement(icon, {size: 'xs'})}
+          <IconOpen {...iconsProps} />
         </IconWrapper>
-      )}
-      {children}
-    </div>
-  )
-)`
+      );
+    }
+
+    return null;
+  }
+
+  function tagWithParent() {
+    if (defined(href)) {
+      return <ExternalLink href={href}>{tag}</ExternalLink>;
+    }
+
+    if (defined(to)) {
+      return (
+        <Link to={to} {...props}>
+          {tag}
+        </Link>
+      );
+    }
+
+    return tag;
+  }
+
+  return <span {...props}>{tagWithParent()}</span>;
+}
+
+const Background = styled('div')<{type: keyof Theme['tag']}>`
   display: inline-flex;
-  box-sizing: border-box;
-  padding: ${p => (p.size === 'small' ? '0.1em 0.4em 0.2em' : '0.35em 0.8em 0.4em')};
-  font-size: ${p => p.theme.fontSizeExtraSmall};
-  line-height: 1;
-  color: ${p => (p.priority ? p.theme.white : p.theme.gray800)};
-  text-align: center;
-  white-space: nowrap;
-  vertical-align: middle;
   align-items: center;
-  border-radius: ${p => (p.size === 'small' ? '0.25em' : '2em')};
-  text-transform: lowercase;
-  font-weight: ${p => (p.size === 'small' ? 'bold' : 'normal')};
-  background: ${p => getPriority(p)?.background ?? p.theme.gray300};
-  ${p => getBorder(p)};
-  ${p => getMarginLeft(p)};
+  height: ${TAG_HEIGHT};
+  border-radius: ${TAG_HEIGHT};
+  background-color: ${p => p.theme.tag[p.type].background};
+  padding: 0 ${space(0.75)};
 `;
 
 const IconWrapper = styled('span')`
   margin-right: ${space(0.5)};
 `;
 
+const Text = styled('span')`
+  color: ${p => p.theme.gray700};
+  font-size: ${p => p.theme.fontSizeExtraSmall};
+  max-width: 150px;
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
```

Comment:
maybe you could use the helper "overflowEllipsis" here alongside "display: inline-block" ?

Reply:
I'd rather not use this as I would have to override 2 of 5 things it does.